differ/differ.go

```diff
@@ -301,6 +309,97 @@ func evaluateFilterExpression(eventFilter string, thresholds EventThresholds, ev
 	return evaluator.Evaluate(eventFilter, values)
 }
 
+func findRenderedReport(reports []types.RenderedReport, ruleName types.RuleName, errorKey types.ErrorKey) (types.RenderedReport, error) {
+	for _, report := range reports {
+		reportRuleName := string(report.RuleID)[strings.LastIndex(string(report.RuleID), ".")+1:]
+		if reportRuleName == string(ruleName) && report.ErrorKey == errorKey {
+			return report, nil
+		}
+	}
+	return types.RenderedReport{}, errors.New("report for given rule ID nad error key has not been found")
+}
+
+func createServiceLogEntry(report types.RenderedReport, cluster types.ClusterEntry) types.ServiceLogEntry {
+	logEntry := types.ServiceLogEntry{
+		ClusterUUID: string(cluster.ClusterName),
+		Description: report.Description,
+		ServiceName: "CCX Notification Service",
+		Summary:     report.Reason,
+	}
+
+	// It is necessary to truncate the fields because of Service Log limitations
```

Comment:
what will happen with the longer values? will the be cuted? maybe we should cross check with PMs if this is the desired behavior

Reply:
Text lengths have been changed according to values mentioned by Pavel.